src/classes/QueryParentPredicate.cls

```diff
@@ -0,0 +1,31 @@
+public class QueryParentPredicate extends QueryPredicate {
```

Comment:
@jamessimone yo dawg, can you back out these changes for now and put a hold on your PR? I'm getting lost trying to keep up with your revisions and I've already got a solution in the works for the parent field filtering